src/goto-programs/goto_program.cpp

```diff
@@ -100,8 +100,20 @@ std::ostream &goto_programt::output_instruction(
     out << '\n';
     break;
 
-  case RETURN:
   case OTHER:
+    if(instruction.get_other().id() == ID_code)
+    {
+      const auto &code = to_code(instruction.get_other());
+      if(code.get_statement() == ID_havoc_object)
+      {
+        out << "HAVOC_OBJECT " << from_expr(ns, identifier, code.op0()) << '\n';
+        break;
+      }
+      // fallthrough
+    }
+    // fallthrough
```

Comment:
⛏️ 💡 might be more readable to just repeat the line from ASSIGN and break.

Reply:
You are right that having:

```c
    out << from_expr(ns, identifier, instruction.get_code()) << '\n';
    break;
```

in place of the two `fallthrough` comments above would probably be more explicit, but this would also lead to code duplication, so I am somewhat hesitant.